k8s/upgrades/jiva-0.6.0-0.8.1/jiva_volume_upgrade.sh

```diff
@@ -0,0 +1,224 @@
+    echo "Replica container not found"; exit 1;
+fi
+
+if [[ -z $c_name ]]; then 
+    echo "Target container not found"; exit 1;
+fi
+
+controller_version=`kubectl get deployment $c_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+
+replica_version=`kubectl get deployment $r_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+
+controller_svc_version=`kubectl get svc $c_svc -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+
+# Get the number of replicas configured. 
+# This field is currently not used, but can add additional validations
+# based on the nodes and expected number of replicas
+rep_count=`kubectl get deploy $r_dep --namespace $ns -o jsonpath="{.spec.replicas}"`
+
+# Get the list of nodes where replica pods are running, delimited by ':'
+rep_nodenames=`kubectl get pods -n $ns \
+ -l "vsm=$pv" -l "openebs/replica=jiva-replica" \
+ -o jsonpath="{range .items[*]}{@.spec.nodeName}:{end}"`
+
+echo "Checking if the node with replica pod has been labeled with $replica_node_label"
+for rep_node in `echo $rep_nodenames | tr ":" " "`; do
+    nl="";nl=`kubectl get nodes $rep_node -o jsonpath="{.metadata.labels.openebs-pv-$pv}"`
+    echo "Labeling $rep_node";
+    kubectl label node $rep_node "openebs-pv-${pv}=$replica_node_label" --overwrite
+done
+
+sed "s/@sc_name@/$sc_name/g" jiva-replica-patch.tpl.json | sed  "s/@sc_resource_version@/$sc_res_ver/g" | sed  "s/@capacity@/$pv_capacity/g" | sed  "s/@replica_node_label@/$replica_node_label/g" | sed  "s/@r_name@/$r_name/g" | sed  "s/@pv-name@/$pv/g" | sed  "s/@target_version@/$target_upgrade_version/g" | sed  "s/@pvc-name@/$pvc_name/g" > jiva-replica-patch.json
+sed "s/@sc_name@/$sc_name/g" jiva-target-patch.tpl.json | sed  "s/@sc_resource_version@/$sc_res_ver/g" | sed  "s/@c_name@/$c_name/g" | sed  "s/@target_version@/$target_upgrade_version/g"  | sed  "s/@pvc-name@/$pvc_name/g" | sed  "s/@pv-name@/$pv/g" | sed  "s/@rep_count@/$rep_count/g" > jiva-target-patch.json
+sed "s/@sc_name@/$sc_name/g" jiva-target-svc-patch.tpl.json | sed  "s/@sc_resource_version@/$sc_res_ver/g" | sed  "s/@target_version@/$target_upgrade_version/g" | sed  "s/@pvc-name@/$pvc_name/g" | sed  "s/@pv-name@/$pv/g" > jiva-target-svc-patch.json
+
+#################################################################################
+# STEP: Patch OpenEBS volume deployments (jiva-target, jiva-replica & jiva-svc) #  
+#################################################################################
+
+#### PATCH JIVA REPLICA DEPLOYMENT ####
+if [[ $replica_version=="" ]] || [[ "$replica_version" != "$target_upgrade_version" ]]; then
+    echo "Upgrading Replica Deployment to $target_upgrade_version"
+
+    # Setting the update stratergy to recreate
+    setDeploymentRecreateStrategy $ns $r_dep
+
+    echo "Remove deprecated lables from Replica Deployment"
+    kubectl patch deployment --namespace $ns $r_dep --type json -p "$(cat replica-patch-remove-labels.json)"
```

Comment:
lets follow the approach that followed in 0.6 to 0.7 upgrade.

Reply:
done moved to the last